src/runtime_src/core/edge/user/aie/common_layer/adf_runtime_api.cpp

```diff
@@ -680,4 +710,224 @@ err_code gmio_api::wait()
+    }
+    XAie_DmaTensor dims = { (uint8_t)dimDescs.size(), dimDescs.data() };
+    driverStatus |= XAie_DmaSetMultiDimAddr(&dmaInst, &dims, bdParam.address, bdParam.length);
+    debugMsg(static_cast<std::stringstream &&>(std::stringstream() << "address " << std::hex << bdParam.address << std::dec << ", length " << bdParam.length).str());
+
+    //zero padding
+    for (int j = 0; j < bdParam.padding.size(); j++)
+    {
+        driverStatus |= XAie_DmaSetZeroPadding(&dmaInst, j, DMA_ZERO_PADDING_BEFORE, bdParam.padding[j].first);
+        driverStatus |= XAie_DmaSetZeroPadding(&dmaInst, j, DMA_ZERO_PADDING_AFTER, bdParam.padding[j].second);
+        debugMsg(static_cast<std::stringstream &&>(std::stringstream() << "D" << j << " zero before " << bdParam.padding[j].first << ", zero after " << bdParam.padding[j].second).str());
+    }
+
+    //packet id
+    if (bdParam.enable_packet)
+    {
+        driverStatus |= XAie_DmaSetPkt(&dmaInst, { .PktId = bdParam.packet_id,.PktType = 0 });
+        driverStatus |= XAie_DmaSetOutofOrderBdId(&dmaInst, bdParam.out_of_order_bd_id);
+    }
+    debugMsg(static_cast<std::stringstream &&>(std::stringstream() << "enable packet " << bdParam.enable_packet << ", packet id " << (uint16_t)bdParam.packet_id << ", out_of_order_bd_id " << (uint16_t)bdParam.out_of_order_bd_id).str());
+
+    //tlast suppress
+    if (bdParam.tlast_suppress)
+        driverStatus |= XAie_DmaTlastDisable(&dmaInst);
+    debugMsg(static_cast<std::stringstream &&>(std::stringstream() << "tlast suppress " << bdParam.tlast_suppress).str());
+
+    //iteration
+    if (bdParam.iteration_stepsize > 0 || bdParam.iteration_wrap > 0 || bdParam.iteration_current > 0)
+        driverStatus |= XAie_DmaSetBdIteration(&dmaInst, bdParam.iteration_stepsize, bdParam.iteration_wrap, bdParam.iteration_current);
+    debugMsg(static_cast<std::stringstream &&>(std::stringstream() << "iteration stepsize " << bdParam.iteration_stepsize << ", iteration wrap " << bdParam.iteration_wrap << ", iteration current " << (uint16_t)bdParam.iteration_current).str());
+
+    //enable compression
+    if (bdParam.enable_compression)
+        driverStatus |= XAie_DmaEnableCompression(&dmaInst);
+    debugMsg(static_cast<std::stringstream &&>(std::stringstream() << "enable compression " << bdParam.enable_compression).str());
+
+    //lock
+    if (bdParam.lock_acq_enable)
+        driverStatus |= XAie_DmaSetLock(&dmaInst, XAie_LockInit(bdParam.lock_acq_id, bdParam.lock_acq_value), XAie_LockInit(bdParam.lock_rel_id, bdParam.lock_rel_value));
+    debugMsg(static_cast<std::stringstream &&>(std::stringstream() << "lock_acq_enable " << bdParam.lock_acq_enable << ", lock_acq_id " << (uint16_t)bdParam.lock_acq_id << ", lock_acq_value " << (int)bdParam.lock_acq_value << ", lock_rel_id " << (uint16_t)bdParam.lock_rel_id << ", lock_rel_value " << (int)bdParam.lock_rel_value).str());
```

Comment:
@stsoe what about a variadic API
```suggestion
    debugMsg("lock_acq_enable ", bdParam.lock_acq_enable, ", lock_acq_id ", static_cast<uint16_t>(bdParam.lock_acq_id), ", lock_acq_value ", static_cast<int>(bdParam.lock_acq_value), ", lock_rel_id ", static_cast<uint16_t>(bdParam.lock_rel_id), ", lock_rel_value ", static_cast<int>(bdParam.lock_rel_value));
```

Reply:
Or just simply use today the Pythonesque https://fmt.dev/latest/index.html while waiting for C++20